### DISCUSSION TITLE
Determining max bonds for given NVS partition size

### DISCUSSION THREAD

**Discussion by thekurtovic** [h2zero/NimBLE-Arduino#595] [Q&A] [answered]
Hello,
I'm using NimBLE for an ESP32 to run as BLE central, but have some questions regarding sdkconfig.
I want NimBLE to save more bonds, but am not sure how much NVS memory is used per bond, so I don't know what I should change the value to. 

1. How many bytes are used when saving a bond in NVS? My NVS partition is 0xD000 bytes in size.
2. If max bonds are increased, should I increase max CCCD's accordingly? (i.e. 3/8 -> 6/16, 9/24, etc.)

The peripherals which the ESP32 connects to will only have 1 service, with two characteristics (one for receiving, one for sending), does this mean I should use a 1:2 ratio for max bonds and CCCDs saved?
Thanks

**Comment by thekurtovic** [comment]
I think I found what I was looking for in ble_store_nvs.c.
It seems to be 88 B per bond and 16 B per CCCD according to the sizeof's from this snippet.
https://github.com/h2zero/NimBLE-Arduino/blob/release/1.4/src/nimble/nimble/host/store/config/src/ble_store_nvs.c#L431-L441

**Comment by h2zero** [comment, marked as answer]
Yes, that's looks correct. This doesn't take a lot of space.
